components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -0,0 +1,653 @@
+        // of the IDP then we can proceed further with the check
+        for (String aud : audience) {
+            // if the audience entry matches with the token end point URL we are good to go
+            if (StringUtils.equals(tokenEndPointAlias, aud)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(tokenEndPointAlias + " of IDP was found in the list of audiences");
+                }
+                audienceFound = true;
+                break;
+            }
+        }
+
+        if (!audienceFound){
+            // we are here because none of the audience(aud) matched the token End Point
+            if (log.isDebugEnabled()) {
+                log.debug("None of the audience values matched the tokenEndpoint Alias " + tokenEndPointAlias);
+            }
+            return false;
+        }
+
+        log.debug("Audience(aud) of the JWT validated successfully");
+
+
+        /**
+         * The JWT MUST contain an exp (expiration) claim that limits the time window during which
+         * the JWT can be used. The authorization server MUST reject any JWT with an expiration time
+         * that has passed, subject to allowable clock skew between systems. Note that the
+         * authorization server may reject JWTs with an exp claim value that is unreasonably far in the
+         * future.
+         */
+        Date expirationTime = claimsSet.getExpirationTime();
+
+        if (expirationTime == null) {
+            log.debug("No Expiration Time(exp) found in the JWT");
+            return false;
+        }
+
+        long currentTimeInMillis = System.currentTimeMillis();
+        long expirationTimeInMillis = expirationTime.getTime();
+        long timeStampSkewMillis = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+
+
+        if (log.isDebugEnabled()) {
+            log.debug("Current time: " + new Date(currentTimeInMillis));
+            log.debug("Expiration Time(exp) found in JWT: " + expirationTime);
+
+            log.debug("Current time in ms: " + currentTimeInMillis);
+            log.debug("Expiration Time(exp) found in JWT in ms: " + expirationTimeInMillis);
+            log.debug("Timestamp skew of the OauthServer : " + timeStampSkewMillis + "ms");
+        }
+
+        // check whether the JWT has expired
+        if ((currentTimeInMillis + timeStampSkewMillis) > expirationTimeInMillis) {
+            if (log.isDebugEnabled()) {
+                log.debug("JSON Web Token : " + signedJWT.getPayload().toJSONObject().toString() + " is expired." +
+                        ", Expiration Time(ms) : " + expirationTimeInMillis +
+                        ", TimeStamp Skew : " + timeStampSkewMillis +
+                        ", Current Time : " + currentTimeInMillis);
+            }
+            log.debug("JWT Rejected and validation terminated");
+            return false;
+        }
+        log.debug("Expiration Time(exp) of JWT was validated successfully");
+
```

Comment:
Excessive debug logs from line #356..  should avoid logging duplicate details that does not reflect additional information